mkosi/burn.py

```diff
@@ -0,0 +1,44 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+import sys
+
+from mkosi.config import (
+    MkosiArgs,
+    MkosiConfig,
+    OutputFormat,
+)
+from mkosi.log import die
+from mkosi.run import run
+
+def run_burn(args: MkosiArgs, config: MkosiConfig) -> None:
+
+    if config.output_format not in (
+        OutputFormat.disk,
+        OutputFormat.esp,
+    ):
+        die(f"{config.output_format} images cannot be dd'ed to disk")
+
+    fname = config.output_dir_or_cwd() / config.output
+
+    if (len(args.cmdline) != 1):
+            die("Expected device argument.");
+
+    device = args.cmdline[0]
+
+    cmd = [
+        "dd",
```

Comment:
While I generally also `dd` things onto devices, I've been told that this is mostly just muscle-memory and a simple `cp` would just work as well and have `cp` figure out the block size and stuff. I have no strong opinion on this, but do you have any thoughts?

Reply:
changed to repart